src/languages/go.js

```diff
@@ -45,14 +45,15 @@ function(hljs) {
       },
       {
         className: 'function',
-        beginKeywords: 'func', end: /\s*\{/, excludeEnd: true,
+        beginKeywords: 'func', end: /\B\b/,
```

Comment:
If this is still goin to be a single line you’d just do {|$

Bracket or end of line.

Reply:
I’d still gobble up the white space first though.

Reply:
Would you still use `\s*`? I'm guessing it won't matter that that includes newlines, as the `$` would presumably match first.

Reply:
I’m not sure it does. Try it and see what the tests do.